src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -153,22 +125,11 @@ public void SetOptions(OptionSet optionSet)
 
                     _currentValues = _currentValues.SetItem(optionKey, setValue);
 
-                    ImmutableArray<Lazy<IOptionSerializer, OptionSerializerMetadata>> optionSerializers;
-                    if (_featureNameToOptionSerializers.TryGetValue(optionKey.Option.Feature, out optionSerializers))
+                    foreach (var serializer in _optionSerializers)
```

Comment:
No, the serializers still live in the above-common layer. The _metadata_ of which options should be serialized where is now metadata on the options themselves.